crates/iceberg/src/spec/schema.rs

```diff
@@ -642,6 +644,199 @@ impl SchemaVisitor for IndexByName {
     }
 }
 
+struct PruneColumn {
+    selected: HashSet<i32>,
+    select_full_types: bool,
+}
+
+impl PruneColumn {
+    fn new(selected: HashSet<i32>, select_full_types: bool) -> Self {
+        Self {
+            selected,
+            select_full_types,
+        }
+    }
+
+    fn project_selected_struct(projected_field: Option<Type>) -> Result<StructType> {
+        match projected_field {
+            // If the field is a StructType, return it as such
+            Some(field) if field.is_struct() => Ok(field.as_struct_type().unwrap_or_default()),
+            // If projected_field is None or not a StructType, return an empty StructType
+            _ => Ok(StructType::default()),
+        }
+    }
+    fn project_list(list: &ListType, result: Option<Type>) -> Result<ListType> {
```

Comment:
We don't need to pass `Option` here?

Reply:
I see! If there are no scenarios where result would be `None`, then there is no need to use `Option<Type>`, right?

Reply:
Yes.